llvm/lib/CodeGen/PreISelIntrinsicLowering.cpp

```diff
@@ -335,6 +335,59 @@ bool PreISelIntrinsicLowering::expandMemIntrinsicUses(Function &F) const {
   return Changed;
 }
 
+static bool lowerExpIntrinsicToLoop(Module &M, Function &F, CallInst *CI) {
```

Comment:
Should generalize this over the intrinsic. Probably should put it in utils

Reply:
Moved it to its own file. Let me know if there's a more appropriate place. Right now, I just generalized it to unary intrinsics